File: drouter-plugin/src/main/groovy/com/didi/drouter/annotation/Remote.java
package com.didi.drouter.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by gaowei on 2020/10/28
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.CLASS)
public @interface Remote {
}

File: plugins/src/main/kotlin/com/didi/drouter/annotation/Remote.kt
package com.didi.drouter.annotation

/**
 * Created by gaowei on 2020/10/28
 */
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER
)
@Retention(
    AnnotationRetention.BINARY
)
annotation class Remote
